packages/suite-desktop/src-electron/modules/tor.ts

```diff
@@ -51,13 +57,13 @@ const init: Module = async ({ mainWindow, store, interceptor }) => {
         mainWindow.webContents.send('tor/status', settings.running);
     };
 
-    ipcMain.on('tor/toggle', async (_, start) => {
+    ipcMain.on('tor/toggle', async (_: unknown, start: boolean) => {
         logger.info('tor', `Toggling ${start ? 'ON' : 'OFF'}`);
```

Comment:
I have a problem that when I'm connected to a backend with Tor on and then I turn it off in Suite, it never reconnects. Based on logs it seems that this `tor/toggle` is called AFTER the `tor/toggle` in `trezor-connect-ipc`, so the proxy there is not unset (because Tor is still running at that time). Do you think it could be related to your changes?

Reply:
Yes the order or things have changed, because now when we start TOR we are waiting until TOR is really available before setting the proxy. It is possible that other things are happening without waiting for that. 

I am not sure if I understand 100% the issue you described.

1. Could you please help me to understand the issue you are describing by providing step by step way to reproduce it?
2. The logs you mentioned are from the suite-desktop or  are you running a backend and you see the logs?

Reply:
Yes, I meant the logs from the suite-desktop. The steps were simply:
1) use Suite with default backends and Tor on
2) turn Tor off from Suite
3) go to any account and you should see the `Backend disconnected, reconnecting` or so message

The other direction was:
1) use Suite with default backends and Tor off
2) turn Tor on from Suite
3) go to coin settings / connection details and you should see that even if there are .onion addresses shown, the real connected backend is not one of them but non-onion instead.

In both cases I suspect that TrezorConnect obtains the Tor changing status event before it's really changed and therefore does nothing.

Reply:
I see the issue, interesting.
Have you tested it on develop? I just did it and the situation is similar, but new develop bran has the message `Backend is disconnected` and it tries to connect but it never connects.

So I do not think the issue is with the changes in this PR, but I will try to investigate further to see how to fix it.

Reply:
At first I agreed that it looks similar on `develop`, but it was just a residual `Tor address and port `settings which persisted between branches. If I reset the app storage, `develop` works as intended.

Reply:
Thank you for your comments, thanks to them I found an issue, now I have to think which is the best way to fix it.

Basically I found out that we are listening to event 'tor/toggle' from `ipcMain` in 2 different places:
* packages/suite-desktop/src-electron/modules/tor.ts
* packages/suite-desktop/src-electron/modules/trezor-connect-ipc.ts

I `modules/tor.ts` we are waiting for TOR to be available and in `modules/trezor-connect-ipc.ts` we are setting connect without waiting for TOR  to be available. And that is causing the issue.

Reply:
@marekrjpolak I added changes so now the issue should not be there anymore. Let me know how it goes in your side!

Reply:
It's resolved, thanks a lot! :pray: I'll re-review the whole thing now.